### PR TITLE
Lesson 03 Files

### PR DESCRIPTION
**Pull request by imx-lk**
I've completed the exercises for lesson 03. 
Thank you

### CODE REVIEW

File: students/imx-lk/lesson03/Slicing_Lab.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+def first_last_exchage(input_string):
+    '''It takes input_string and returns the first and last items exchanged'''
+    #I decided to save my input_string into a different varibable because I want to check the data type on line 26 and I am planning on turning this into a list
+
+    split_firstlast_input = input_string
+    #make it into a list to easly change the first and last items
+    list_first_last= list(split_firstlast_input)
+    #savenice  the first and last letters of the string
+    first = split_firstlast_input [0]
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/Slicing_Lab.py:17]: Remove whitespace before bracket - here and below

File: students/imx-lk/lesson03/Slicing_Lab.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    #First check what type of item it is
+    data_type = type(input_string)
+
+    if data_type == str:
+        convert_to_string = "".join(list_first_last)
+        return convert_to_string
+
+    if data_type == tuple:
+        convert_to_tuple =tuple(list_first_last)
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/Slicing_Lab.py:34]: Missing whitespace after assignment operator

File: students/imx-lk/lesson03/Slicing_Lab.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+if __name__ == "__main__":
+    # run some tests against exchange_fist_last()
+    a_string = 'this is a string'
+    a_tuple = (2, 54, 13, 12, 5, 32)
+    a_list = [12,45, 'osito', 'ilsse']
+
+    #Need a longer sequences to test the_four
+    a_string_longer = 'this is a string that we will remove the first four and the last four and then every other word'
+    a_tuple_longer = (12,34,56,78,90,3,4,5,6,7,8,9)
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/Slicing_Lab.py:91]: You want whitespace after every comma here

File: students/imx-lk/lesson03/Slicing_Lab.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    assert first_last_exchage(a_list) == ['ilsse', 45, 'osito', 12]
+    print('First function passed')
+
+    assert every_other(a_string) == 'ti sasrn'
+    assert every_other(a_tuple) == (2, 13, 5)
+    assert every_other(a_list) == [12, 'osito']
+    print('Second function passed')
+
+    assert the_four(a_string_longer) =='that will the four the four'
+    assert the_four(a_tuple_longer) == (90,4)
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/Slicing_Lab.py:106]: Whitespace needed after this comma too

File: students/imx-lk/lesson03/Slicing_Lab.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    assert reversed_items(a_string) =='gnirts a si siht'
+    assert reversed_items(a_tuple) == (32, 5, 12, 13, 54, 2)
+    assert reversed_items(a_list) == ['ilsse', 'osito', 45, 12]
+    print('Fourth function passed')
+
+    assert the_thirds(a_string) == 'stringthis is a '
+    assert the_thirds(a_tuple) == (5, 32, 2, 54, 13, 12)
+    assert the_thirds(a_list_longer) == ['mantra', 88, 63, 89, 56, 12, 45, 'osito', 'ilsse', 34, 78, 89, 'Damien']
+    print('ALL PASSED!')
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/Slicing_Lab.py:118]: Works well!

File: students/imx-lk/lesson03/list_lab.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+print('\n----Series 1 ----')
+fruit =['Apples', 'Pears', 'Oranges', 'Peaches']
+print(fruit)
+response = input("Enter your favorite fruit and we will add it to the list: ")
+fruit.append(response.capitalize())
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/list_lab.py:7]: Nice use of capitalize

File: students/imx-lk/lesson03/list_lab.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+print(fruit)
+
+#Display the fruit according to the index
+response_num = int(input("Enter a number and I will display the fruit (1st item is # 1): "))
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/list_lab.py:11]: len() is nice to use here to show the user how long the list is

File: students/imx-lk/lesson03/mailroom_pt1.py [hunk 8]
```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+#The original list of donors will sit outside the main function because if it stays inside the fucntion it will replace the names appended in thankyou_new_donor
+donor_list = [['Damien Leake', 1000, 3000, 500], ['Andrew Foreman', 375, 2900], ['David Altman', 20, 30000, 54678.50], ['Alex Haldin', 1.75, 123678,],['Osito Castaneda', 123456.75, 34567.12, 35]]
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/mailroom_pt1.py:3]: Missing some whitespaces after commas

File: students/imx-lk/lesson03/mailroom_pt1.py [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+
+def create_report(donor_list):
+#this list will capture the total, average, and number of gifts
+    list_total_avg = []
+
+#This for loop grabs the first list within donor_list
+    for donor_entry in donor_list:
+        #Need to clear out the total and average everytime or it adds up donations continuesly
+        total_donations= 0
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/mailroom_pt1.py:108]: Missing whitespace here

File: students/imx-lk/lesson03/mailroom_pt1.py [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+
+        #Call the function according to which option in the menu they picked
+        if response == '1':
+            thankyou_note(donor_list)
+        elif response == '2':
+            create_report(donor_list)
+        else:
+            print('\nGoodbye thank you for using the mailroom!')
+            break
+        continue
```
Comment (charlesrobison) [inline students/imx-lk/lesson03/mailroom_pt1.py:156]: A couple things missing here.  1) Report should be sorted by total given in descending order.  2) When adding new donations, it should add to the total for a donor, but it's currently creating duplicate donors.

**Comment by natasha-aleksandrova** [comment]
@imx-lk can you please try syncing your fork with master?

**Comment by natasha-aleksandrova** [comment]
@imx-lk we're going to go ahead and close this for now since we're waiting for you to sync the files. your grade should have been entered regardless 🙂

**Pull request closed without merging**